src/js/tracing/nativeframes.ts

```diff
@@ -0,0 +1,263 @@
+import { Span, Transaction } from "@sentry/tracing";
+import { Event, EventProcessor, Measurements } from "@sentry/types";
+import { logger, timestampInSeconds } from "@sentry/utils";
+
+import { NativeFramesResponse } from "../definitions";
+import { NATIVE } from "../wrapper";
+import { instrumentChildSpanFinish } from "./utils";
+
+interface FramesMeasurements extends Measurements {
+  frames_total: {
+    value: number;
+  };
+  frames_slow: {
+    value: number;
+  };
+  frames_frozen: {
+    value: number;
+  };
+}
+
+const MARGIN_OF_ERROR_SECONDS = 0.05;
+
+/**
+ * Instrumentation to add native slow/frozen frames measurements onto transactions.
+ */
+export class NativeFramesInstrumentation {
+  private _finishFrames: Map<
+    string,
+    { timestamp: number; nativeFrames: NativeFramesResponse | null }
+  > = new Map();
+  private _framesListeners: Map<string, () => void> = new Map();
+  private _lastSpanFinishFrames?: {
+    timestamp: number;
+    nativeFrames: NativeFramesResponse;
+  };
```

Comment:
Would appreciate some comments about what these data structures are, so it's easier for future readers/users.